### PR TITLE
Code quality: Seal all non-inherited classes to improve performance

### LINKED ISSUE #12304: Code Quality: Seal classes when possible
### Description

Related to #4180.

Unsealed classes introduce extra overhead for running each virtual method/type checks because the compiler/JIT has to do extra inheritance checks.

For example, there are three non-sealed classes called `ClassA`, `ClassB` and `ClassC`, and `ClassB` inherits from `ClassA`. If `ClassC` doesn't inherit from `ClassB` (in other forms, if nothing inherits from `ClassB`) then we don't need the extra dispatch operations or checks. Thus making `ClassB` sealed removes that overhead when calling virtual methods or doing type checks.

See https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-6/#peanut-butter for more in-depth details.

### Concerned code

Most C# source files in the entire app solution.

### Gains

- Slightly better performance.

### Requirements

- Make classes that don't need to be inherited sealed.

### Comments

_No response_

### PR DESCRIPTION
**Pull request by gumbarros**
**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
Closes #12304

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?

### CODE REVIEW

**Review by yaira2 — DISMISSED** [review]

File: src/Files.App/Helpers/WMI/ManagementEventWatcher.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ namespace Files.App.Helpers
 	public delegate void EventArrivedEventHandler(object sender, EventArrivedEventArgs e);
 
 	/// <summary>
-	/// A public class used to start/stop the subscription to specific indication source,
+	/// A public sealed class used to start/stop the subscription to specific indication source,
```
Comment (0x5bfa) [inline src/Files.App/Helpers/WMI/ManagementEventWatcher.cs:14]: Why is this changed?

**Review by 0x5bfa — COMMENTED** [review]
lgtm.

File: src/Files.App/Helpers/WMI/ManagementEventWatcher.cs (hunk 1 above)
Comment (gumbarros) [reply src/Files.App/Helpers/WMI/ManagementEventWatcher.cs:14]: Good eye, I changed the classes using Replace All, compiling and seeing what can be `sealed` or not.

Comment (0x5bfa) [reply src/Files.App/Helpers/WMI/ManagementEventWatcher.cs:14]: I'd do that too but is that necessary change or does it make sense to be changed?

Comment (gumbarros) [reply src/Files.App/Helpers/WMI/ManagementEventWatcher.cs:14]: It's not a necessary change, ManagementEventWatcher is not sealed.

**Review by 0x5bfa — APPROVED** [review]

**Comment by yaira2** [comment]
I think we should merge this regardless, but it would be interesting to see if there is a performance difference.
